File: typespec/hub/education.go
package hub

import "github.com/psankar/vetchi/typespec/common"

type AddEducationRequest struct {
	InstituteDomain common.Domain `json:"institute_domain" validate:"required,validate_domain"`
	Degree          string        `json:"degree"           validate:"required,min=3,max=64"`
	StartDate       *string       `json:"start_date"       validate:"omitempty,validate_date,no_future_date,required_with=EndDate"`
	EndDate         *string       `json:"end_date"         validate:"omitempty,validate_date,date_after=StartDate"`
	Description     *string       `json:"description"      validate:"omitempty,max=1024"`
}

type AddEducationResponse struct {
	EducationID string `json:"education_id"`
}

type FilterInstitutesRequest struct {
	Prefix string `json:"prefix" validate:"required,min=3,max=64"`
}

type DeleteEducationRequest struct {
	EducationID string `json:"education_id" validate:"required,uuid"`
}

type ListEducationRequest struct {
	UserHandle *common.Handle `json:"user_handle"`
}

File: typespec/hub/education.ts
export interface AddEducationRequest {
  institute_domain: string;
  degree?: string;
  start_date?: string;
  end_date?: string;
  description?: string;
}

export interface AddEducationResponse {
  education_id: string;
}

export interface Institute {
  domain: string;
  name: string;
}

export interface FilterInstitutesRequest {
  prefix: string;
}

export interface DeleteEducationRequest {
  education_id: string;
}

export interface Education {
  id?: string;
  institute_domain: string;
  degree?: string;
  start_date?: string;
  end_date?: string;
  description?: string;
}

export interface ListEducationRequest {
  user_handle?: string;
}

export interface FilterInstitutesResponse {
  // Maximum 10 institutes will be returned in random order.
  institutes: Institute[];
}
